certbot/certbot/_internal/client.py

```diff
@@ -454,7 +454,22 @@ def obtain_certificate(self, domains: List[str], old_keypath: Optional[str] = No
                 os.remove(csr.file)
             return self.obtain_certificate(successful_domains)
         else:
-            cert, chain = self.obtain_certificate_from_csr(csr, orderr)
+            try:
+                cert, chain = self.obtain_certificate_from_csr(csr, orderr)
+            except messages.Error as error:
```

Comment:
It seems to me that `--allow-subset-of-names` should also be expanded to work on order creation. 

For example, in the case where one of the domains in the order is suddenly forbidden by CA policy, or if the CA decides to additionally check CAA records at order creation time.

To achieve this, the call to `_get_order_and_authorizations` also needs to be part of the retrying logic.

Reply:
Refactored the retry logic into `_retry_obtain_certificate` to make it reusable and clean up `obtain_certificate`.

Now using this to handle acme errors from `_get_order_and_authorizations` and `obtain_certificate_from_csr`, and added an additional test for retrying on a failure during `_get_order_and_authorizations`.